Test/comp/Collections.dfy

```diff
@@ -49,14 +50,109 @@ method Sets() {
 
 // -------------------------------------------------------------------------------------------
 
+abstract module Order {
+  type T(==)
+  method lt(a: T, b: T) returns (r: bool)
+}
+
+module CharOrder refines Order {
+  type T = char
+  method lt(a: T, b: T) returns (r: bool) {
+    r := a < b;
+  }
+}
+
+module CharSetSetOrder refines Order {
+  type T = set<char>
+
+  function method Pow2(i: nat) : nat {
+    if i == 0 then 1 else 2 * Pow2(i-1)
+  }
+
+  method SetAsInt(s: set<char>) returns (r: int) {
+    var ss := s;
+    var acc := 0;
+    while ss != {} {
+      var i: char :| i in ss;
+      ss := ss - {i};
+      acc := acc + Pow2(i as nat);
+    }
+    r := acc;
+  }
+
+  method lt(a: T, b: T) returns (r: bool) {
+    var aInt := SetAsInt(a);
+    var bInt := SetAsInt(b);
+    r := |a| < |b| || (|a| == |b| && aInt <= bInt);
+  }
+}
+
+abstract module PrintOrderedSet {
```

Comment:
This is nice btw, perhaps we can extract this into a shared location in the test suite later.